docs/content/tutorial/step_07.ngdoc

```diff
@@ -1,380 +1,311 @@
-URL. All variables defined with the `:` notation are extracted into the
-{@link ngRoute.$routeParams `$routeParams`} object.
+      ctrl = $componentController('phoneList', {$scope: {}});
+    }));
 
+    ...
 
-## Controllers
+  });
 
-__`app/js/controllers.js`:__
+});
+```
 
-```js
-var phonecatControllers = angular.module('phonecatControllers', []);
+<div class="alert alert-info">
+  **Note:** Because we loaded Jasmine and `angular-mocks.js` in our test environment, we got two
+  helper methods {@link angular.mock.module module} and {@link angular.mock.inject inject} that we
+  can use to access and configure the injector.
+</div>
 
-phonecatControllers.controller('PhoneListCtrl', ['$scope', '$http',
-  function ($scope, $http) {
-    $http.get('phones/phones.json').success(function(data) {
-      $scope.phones = data;
-    });
+We created the controller in the test environment, as follows:
 
-    $scope.orderProp = 'age';
-  }]);
+* We used the `inject()` helper method to inject instances of
+  {@link ngMock.$componentController $componentController} and {@link ng.$httpBackend $httpBackend}
+  services into Jasmine's `beforeEach()` function. These instances come from an injector which is
+  recreated from scratch for every single test. This guarantees that each test starts from a well
+  known starting point and each test is isolated from the work done in other tests.
 
-phonecatControllers.controller('PhoneDetailCtrl', ['$scope', '$routeParams',
-  function($scope, $routeParams) {
-    $scope.phoneId = $routeParams.phoneId;
-  }]);
-```
+* We called the injected `$componentController` function passing the name of the `phoneList`
+  component (whose controller we wanted to instantiate) as a parameter.
 
-Again, note that we created a new module called `phonecatControllers`. For small AngularJS
-applications, it's common to create just one module for all of your controllers if there are just a
-few. As your application grows, it is quite common to refactor your code into additional modules.
-For larger apps, you will probably want to create separate modules for each major feature of
-your app.
+Because our code now uses the `$http` service to fetch the phone list data in our controller, before
+we create the `PhoneListController`, we need to tell the testing harness to expect an incoming
+request from the controller. To do this we:
 
-Because our example app is relatively small, we'll just add all of our controllers to the
-`phonecatControllers` module.
+* Inject the `$httpBackend` service into the `beforeEach()` function. This is a
+  {@link ngMock.$httpBackend mock version} of the service that in a production environment
+  facilitates all XHR and JSONP requests. The mock version of this service allows us to write tests
+  without having to deal with native APIs and the global state associated with them — both of which
+  make testing a nightmare.
```

Comment:
And also overcomes the asynchronous nature of these calls, which slows down unit tests